Allow to preserve quotes

missingcharacter opened the pull request:
<!-- Describe what the change is, trying to link it with open issues -->

### Preserve Quotes

Default: `preserve_quotes: bool = False`<br>
Environment variable override:
```bash
export YAMLFIX_preserve_quotes="false"
```

Keep quotes as they are

```yaml
# original
key: [value, 'value with spaces']
```

```yaml
# Option set to: false
key: [value, value with spaces]
```

```yaml
# Option set to: true
key: [value, 'value with spaces']
```

**Notes:**

You may NOT want to use it in combination with [quote basic values](#quote-basic-values) or [quote keys and basic values](#quote-keys-and-basic-values), because the output may not be uniform

Example: `quote_basic_values`

```yaml
# original
key: [value, 'value with spaces']
```

```yaml
# preserve_quotes = true
# quote_basic_values = true
# quote_representation = "
key: ["value", 'value with spaces']
```

Example: `quote_keys_and_basic_values`

```yaml
# original
key: [value, 'value with spaces']
```

```yaml
# preserve_quotes = true
# quote_keys_and_basic_values = true
# quote_representation = "
"key": ["value", 'value with spaces']
```

## Checklist

* [X] Add test cases to all the changes you introduce
* [X] Update the documentation for the changes

lyz-code commented on docs/index.md:
````diff
@@ -429,6 +429,31 @@ key: 'value'
+
+```yaml
+# Option set to: true
+key: [exec, 'value with spaces']
````
Shouldn't this line be?

```yaml
key: [value, 'value with spaces']
```

So `exec` -> `value`.

missingcharacter replied:
I fixed this typo, thanks 🙇🏼

lyz-code reviewed (changes requested):
Hi @missingcharacter thank you for taking the time to create the PR. To be able to merge it though I think we need to see how does this option work with other existent quote configurations:

* [Quote Basic Values](https://lyz-code.github.io/yamlfix/#quote-basic-values)
* [Quote Keys and Basic Values](https://lyz-code.github.io/yamlfix/#quote-keys-and-basic-values)
* [Quote Representation](https://lyz-code.github.io/yamlfix/#quote-representation)

Maybe there are undesired effects. For each of them, if they do work together I'd add a test, if they don't I'd add a note in the documentation telling the user that this flag doesn't work well with X or Y configuration option.

Thanks :)

missingcharacter commented:
Hi @lyz-code I added tests combining `preserve_quotes` with `quote_basic_values` and `quote_keys_and_basic_values`

I also added a note on the documentation showing the behavior when using `preserve_quotes` in combination with `quote_basic_values` and `quote_keys_and_basic_values`

lyz-code commented:
Thank you so much for implementing the requested changes and for doing the PR

lyz-code commented:
Available since `1.9.0`

missingcharacter commented:
Thank you @lyz-code 🙇🏽‍♂️
